benchmarking/compute_probability_of_improvement.py

```diff
@@ -0,0 +1,251 @@
+                    f" to specify which algorithms runs to compare.",
+                )
+
+        if args.baseline_algo is None:
+            if len(baseline_algos) == 1:
+                args.baseline_algo = baseline_algos[0]
+            elif args.algo in baseline_algos:
+                args.baseline_algo = args.algo
+            else:
+                raise ValueError(
+                    f"The baseline run directory [{args.baseline_runs_dir}] contains "
+                    f"runs for the algorithms [{', '.join(baseline_algos)}]. "
+                    f"Please use the --baseline-algo option specify which one to "
+                    f"compare to.",
+                )
+
+        if args.algo not in algos:
+            raise ValueError(
+                f"The run directory [{args.runs_dir}] contains runs for the algorithms"
+                f" [{', '.join(algos)}]. You specified [{args.algo}], for which no"
+                f" runs can be found in the run directory",
+            )
+
+        if args.baseline_algo not in baseline_algos:
+            raise ValueError(
+                f"The baseline run directory [{args.baseline_runs_dir}] contains runs "
+                f"for the algorithms [{', '.join(baseline_algos)}]. "
+                f"You specified [{args.baseline_algo}], for which no runs can be found"
+                f" in the baseline run directory",
+            )
+
+        if (args.algo == args.baseline_algo) and (
+            args.runs_dir == args.baseline_runs_dir
+        ):
+            warnings.warn(
+                "You are comparing two equal sets of runs. "
+                "This is probably not what you want.",
+            )
+
+        envs = runs_by_algo_and_env[args.algo].keys()
+        baseline_envs = baseline_runs_by_algo_and_env[args.baseline_algo].keys()
+
+        comparison_envs = set(envs).intersection(set(baseline_envs))
+
+        if len(comparison_envs) == 0:
+            raise ValueError(
+                f"The baseline runs are for the environments "
+                f"[{', '.join(baseline_envs)}], while the runs are for the "
+                f"environments [{', '.join(envs)}]. "
+                f"There is no overlap in the environments of the two run sets, so no "
+                f"comparison can be made",
+            )
+
+        ignoring_some_envs = len(comparison_envs) < len(envs)
+        ignoring_some_baseline_envs = len(comparison_envs) < len(baseline_envs)
+        if ignoring_some_envs or ignoring_some_baseline_envs:
+            warnings.warn(
+                f"The baseline runs are for the environments "
+                f"[{', '.join(baseline_envs)}], "
+                f"while the runs are for the environments [{', '.join(envs)}]. "
+                f"The comparison will only be made for the environments "
+                f"[{', '.join(comparison_envs)}].",
+            )
+
+    except ValueError as e:
```

Comment:
Why us this handler instead of letting it dump a normal traceback? Normal traceback seems more useful.

Reply:
I found it easier to read but you are probably right about it. I will revert it.